test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/TimeoutMock.scala

```diff
@@ -0,0 +1,110 @@
+package org.scalajs.testsuite.jsinterop
+
+import scala.annotation.tailrec
+import scala.collection.immutable.TreeMap
+import scala.scalajs.js
+import scala.scalajs.js.annotation.ScalaJSDefined
+
+object TimeoutMock {
+
+  import js.Dynamic.global
+
+  private val realSetTimeout =
+    global.setTimeout.asInstanceOf[js.Function]
+
+  private val realClearTimeout =
+    global.clearTimeout.asInstanceOf[js.Function]
```

Comment:
This is not very composable. These `val`s should be saved as local vals in `withMockedTimeout`